plugins/modules/digital_ocean_droplet.py

```diff
@@ -218,13 +222,21 @@ def __init__(self, module):
         self.unique_name = self.module.params.pop('unique_name', False)
         # pop the oauth token so we don't include it in the POST data
         self.module.params.pop('oauth_token')
+        self.id = None
+        self.name = None
+        self.size = None
+        self.status = None
 
     def get_by_id(self, droplet_id):
         if not droplet_id:
             return None
         response = self.rest.get('droplets/{0}'.format(droplet_id))
         json_data = response.json
         if response.status_code == 200:
+            self.id = json_data['droplet']['id']
```

Comment:
Similar to my comments on your previous PR, for sections of code like this, consider writing code like this instead:

```
droplet = json_data.get('droplet', None)
if droplet is not None:
    self.id = droplet.get('id', None)
    self.name = droplet.get('name', None)
    self.size = droplet.get('size_slug', None)
    self.status = droplet.get('status', None)
```

Using the `.get()` method on the dictionary allows you to specify a default value if the value is not found, rather than a direct index (which would make the code explode with a `KeyError`).

Doing it this way is less verbose than the `dictionary is not None and key in dictionary:` way that I had suggested before, and makes the code more defensive against weird/unexpected JSON returns.

Reply:
> Similar to my comments on your previous PR, for sections of code like this, consider writing code like this instead:
> 
> ```
> droplet = json_data.get('droplet', None)
> if droplet is not None:
>     self.id = droplet.get('id', None)
>     self.name = droplet.get('name', None)
>     self.size = droplet.get('size_slug', None)
>     self.status = droplet.get('status', None)
> ```
> 
> Using the `.get()` method on the dictionary allows you to specify a default value if the value is not found, rather than a direct index (which would make the code explode with a `KeyError`).
> 
> Doing it this way is less verbose than the `dictionary is not None and key in dictionary:` way that I had suggested before, and makes the code more defensive against weird/unexpected JSON returns.

Ah, this is much nicer looking, thanks Mike -- will fix.